File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string stack;

        for (char c : s) {
            if (!stack.empty() && stack.back() == c) {
                stack.pop_back();  // remove the adjacent duplicate
            } else {
                stack.push_back(c);  // add the character
            }
        }

        return stack;
    }
};

File: 1128-remove-all-adjacent-duplicates-in-string/remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []

        for char in s:
            if stack and stack[-1] == char:
                stack.pop()  # remove the adjacent duplicate
            else:
                stack.append(char)  # add the character

        return ''.join(stack)